### PR TITLE
DEV-648 - Login Form + Filtered Selection component

### PR DESCRIPTION
**Pull request by respinos**
Should be considered adding more raw materials for future development:

* adding `js/components` to collect reusable components, e.g. `FilterableSelection.svelte`
* as opposed to svlete components that are more views? e.g. `LoginForm.svelte`
* updated `main.js`
    * import components that should be made available to HTDL applications
    * instantiate components using custom elements (e.g. `hathi-echo`) or on buttons that will trigger their display
* documented some experiments that probably should not see the light of day
* updated `vite.config.js` to publish all the HTML pages in `src`

**TO SEE THE LOGIN FORM**

1. `npm run start`
2. open http://localhost:5173/navbar_login_example.html

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,4 +24,30 @@ Icons currently are included via a `<script>` call, due to [this issue](https://
...
+The `navbar_login_example.html` has a NOP handler for invoking the `ping` utility:
+
+```html
+  <script>
+    function ping_handler(status) {};
+  </script>
+  <script src="https://babel.hathitrust.org/cgi/ping"></script>
+```
+
+`ping` returns a JSONP response, invoking `ping_handler()` and the `LoginForm` component checks for the presence of `HT.login_status` to build its institution list.
```
Comment (carylwyatt) [inline README.md:53]: omg thank you for explaining this! when I was looking at `ping` during feedback-form creation, I was like "why... is every institution listed in here??" glad to know there's a reason 😂

File: src/js/LoginForm.svelte
Comment (carylwyatt) [inline src/js/LoginForm.svelte:1]: This is more of an organizational question: what constitutes a "component"? 
For example, why is LoginForm under `/src/js` but FilterableSelection is in `/components`?
I'm setting up spec/test files for storybook and had a similar question about how to organize the individual pieces of all this.

**Review by carylwyatt — COMMENTED** [review]
Just the one question about where things "belong," but otherwise, this all looks great! I hadn't really thought through using slots and didn't really have any idea how this was all going to integrate into the apps, so it's cool to see it coming along.

File: src/js/LoginForm.svelte
Comment (respinos) [reply src/js/LoginForm.svelte:1]: I don't have a good answer, and we can totally rearrange the files. At the moment, I saw `components` as containing the re-usable components, and everything in `src/js` as more a page? e.g. nothing is going to re-use `LoginForm`. But this all gets flatted in `main.js` so ¯\_(ツ)_/¯

Comment (respinos) [reply src/js/LoginForm.svelte:1]: Let's go with whatever makes sense for the storybook effort. I'd lean toward svelete code being in `components` to separate them from `lib` and `main.js` but that's not a hard preference. Let me know and I'll update the PR.

Comment (carylwyatt) [reply src/js/LoginForm.svelte:1]: I agree, let's put all the svelte pieces in `components`. The story files will be adjacent to the component being tested. Hopefully it doesn't get too cluttered in there, but I guess we can always drill down if things get out of control.

Comment (respinos) [reply src/js/LoginForm.svelte:1]: OK --- that's done. Cleaned up some earlier experiments.

**Pull request merged**